mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/options/MongoDBCollection.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.mongodb.options;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.hibernate.ogm.datastore.mongodb.options.impl.MongoDBCollectionConverter;
+import org.hibernate.ogm.options.spi.MappingOption;
+
+/**
+ * MongoDB specific definitions for a collection.
+ *
+ * @author Guillaume Smet
+ */
+@Target({ TYPE, METHOD, FIELD })
+@Retention(RUNTIME)
+@MappingOption(MongoDBCollectionConverter.class)
+public @interface MongoDBCollection {
```

Comment:
We can discuss that, but it'd be quite a shift from the principles OGM follows so far. We are looking to provide a unified programming model, finding useful abstractions as far as possible. What you describe essentially is more to have an "OGM for MongoDB", an "OGM for Cassandra" etc.

Also there is the case of polyglot persistence where an entity is mapped to several stores. Being able to re-use config here would be nice instead of having `@MongoCollection`, `@CassandraCollection` etc. all there.

That said, I see your point about being unsure of what's supported on a given store or not. But that's a general issue, well, challenge ;), with OGM.

@emmanuelbernard, what's your take?